CoreWidgetProvider/Helpers/ChartHelper.cs

```diff
@@ -1,138 +1,226 @@
+            // * a grey box, outlining the entire image
+            // The SVG also contains a definition for the fill gradient.
+            var svgElement = CreateBlankSvg(height, width);
+
+            // Create the line that will show the points on the graph.
+            var lineElement = new XElement("polyline");
+            var points = TransformPointsToLine(chartValues, out var startX, out var finalX);
+            lineElement.SetAttributeValue("points", points.ToString());
+            lineElement.SetAttributeValue("style", GetLineStyle(type));
+
+            // Create the line that will contain the gradient fill.
+            TransformPointsToLoop(points, startX, finalX);
+            var fillElement = new XElement("polyline");
+            fillElement.SetAttributeValue("points", points.ToString());
+            fillElement.SetAttributeValue("style", FillStyle);
+
+            // Add the gradient definition and the three shapes to the svg.
+            svgElement.Add(CreateGradientDefinition(type));
+            svgElement.Add(fillElement);
+            svgElement.Add(lineElement);
+            svgElement.Add(CreateBorderBox(height, width));
+
+            chartDoc.Add(svgElement);
+        }
+
+        return chartDoc.ToString();
+    }
+
+    private static XElement CreateBlankSvg(int height, int width)
+    {
+        var svgElement = new XElement("svg");
+        svgElement.SetAttributeValue("height", height);
+        svgElement.SetAttributeValue("width", width);
+        return svgElement;
+    }
+
+    private static XElement CreateGradientDefinition(ChartType type)
+    {
+        var defsElement = new XElement("defs");
+        var gradientElement = new XElement("linearGradient");
+
+        // Vertical gradients are created when x1 and x2 are equal and y1 and y2 differ.
+        gradientElement.SetAttributeValue("x1", "0%");
+        gradientElement.SetAttributeValue("x2", "0%");
+        gradientElement.SetAttributeValue("y1", "0%");
+        gradientElement.SetAttributeValue("y2", "100%");
+        gradientElement.SetAttributeValue("id", "gradientId");
+
+        string stop1Style;
+        string stop2Style;
+        switch (type)
+        {
+            case ChartType.GPU:
+                stop1Style = GPUBrushStop1Style;
+                stop2Style = GPUBrushStop2Style;
+                break;
+            case ChartType.Mem:
+                stop1Style = MemBrushStop1Style;
+                stop2Style = MemBrushStop2Style;
+                break;
+            case ChartType.Net:
+                stop1Style = NetBrushStop1Style;
+                stop2Style = NetBrushStop2Style;
+                break;
+            case ChartType.CPU:
+            default:
+                stop1Style = CPUBrushStop1Style;
+                stop2Style = CPUBrushStop2Style;
+                break;
+        }
+
+        var stop1 = new XElement("stop");
+        stop1.SetAttributeValue("offset", "0%");

```

Comment:
consider making these properties `const` fields so you can reference all of them by symbol rather than by string at each callsite